fix: add handling of dns multiaddrs + bootstrapping + CLI / Conn changes (#547)
ipvm-wg/homestar: examples/websocket-relay/README.md

## Tips & Common Issues

- On macOS, for example, a simple homebrew install would install everything you
  need: `brew install rust npm ipfs`.

> - Running `homestar` via `cargo run` here requires a minimum Rust version of
>   `1.73.0`. If you've got an older install of rust, update it with
>   `rustup update`.

Reviewer: ```suggestion
- Running `homestar` using `cargo run` requires a minimum Rust version of
  `1.73.0`. If you've got an older version of rust, update it with
  `rustup update`.
```

- You do not have to start Kubo (IPFS) on your own. The example will do this
  for you, and use `examples/websocket-relay/tmp/.ipfs` as a local blockstore.
  Feel free to discard it when you don't need it.

- If you're already running an IPFS instance, for example [IPFS Desktop][ipfs-desktop],
  the application will check for it and not start a new, local one.
  However, the application expects a default IPFS host and port. The expected
  IPFS `host` and `port` can be updated in the `homestar` network settings:

  ``` toml
  [node]

  [node.network.ipfs]
  host = "127.0.0.1"
  port = 5001
  ```

- We have officially packaged homestar binaries using brew, so
  `brew install fission-codes/fission/homestar` will install mostly everything
  you need, including `ipfs`. You will still need `npm` to run this example, and

>   you'll have to manually `ipfs add` the `synthcat.ping` and `example_test.wasm`

Reviewer: ```suggestion
  you'll have to manually `ipfs add` the `synthcat.png` and `example_test.wasm`
```

>   files located in this directory. Then, from this folder, you can then run the
>   example like this:

Reviewer: ```suggestion
  files located in this directory. Then, from this folder, you can run the
  example like this:
```

  ```
  homestar start --db homestar.db
  ```

>   Afterward, as mentioned above, you can then run `npm install --prefix relay-app` to
>   install dependencies and `npm run --prefix relay-app dev` to start the
>   relay web application (UI) on `http://localhost:5173/` by default.

Reviewer: ```suggestion
  Afterward, run `npm install --prefix relay-app` to install dependencies
  and `npm run --prefix relay-app dev` to start the relay web
  application (UI) on `http://localhost:5173/` by default.
```
